events/src/utils.py

```diff
@@ -313,6 +314,40 @@ def plugin_webhook(message):
     webhooks_pool.send(message["event"], message["ts"], message_json)
 
 
+class NamedTempOpen(object):
+    """
+        This class is used to create a temporary file which is then written to with contents.
+        The temp file is then persisted with the give name by calling os.rename().
+        This class is used to avoid the data loss or truncation in case of multiple processes
+        writing to the same file without the use of fcntl locks.
+
+        The temporary file is created in the cwd of the process.
+    """
+
+    def __init__(self, filename, open_mode, *args, **kwagrs) -> None:
+        self.filename = filename
+        self.open_mode = open_mode
+        self.open_args = args
+        self.open_kwargs = kwagrs
+        self.fileobj = None
+
+    def __enter__(self):
+        tfile = NamedTemporaryFile(mode=self.open_mode,
+                                   delete=False,
+                                   dir=".",
+                                   *self.open_args,
+                                   **self.open_kwargs)
+
+        if tfile is None:
+            raise
```

Comment:
This can only be done inside a exception handler block, otherwise you need to specify the exception.

Reply:
Missed that. Fixed.